From 63ae1903c0142511a0079441fb3880e1ad0ceb21 Mon Sep 17 00:00:00 2001
From: Lee Jackson <info@arducam.com>
Date: Fri, 26 Aug 2022 11:41:49 +0800
Subject: [PATCH 0387/1002] media: i2c: arducam-pivariety: Add custom controls

Add support for strobe_shift, strobe_width and mode custom controls.

Signed-off-by: Lee Jackson <info@arducam.com>
---
 drivers/media/i2c/arducam-pivariety.c | 6 ++++++
 drivers/media/i2c/arducam-pivariety.h | 3 +++
 2 files changed, 9 insertions(+)

diff --git a/drivers/media/i2c/arducam-pivariety.c b/drivers/media/i2c/arducam-pivariety.c
index 5ef8b212ba49..6bb9e9c48e5c 100644
--- a/drivers/media/i2c/arducam-pivariety.c
+++ b/drivers/media/i2c/arducam-pivariety.c
@@ -1140,6 +1140,12 @@ static const char *pivariety_ctrl_get_name(u32 id)
 		return "trigger_mode";
 	case V4L2_CID_ARDUCAM_IRCUT:
 		return "ircut";
+	case V4L2_CID_ARDUCAM_STROBE_SHIFT:
+		return "strobe_shift";
+	case V4L2_CID_ARDUCAM_STROBE_WIDTH:
+		return "strobe_width";
+	case V4L2_CID_ARDUCAM_MODE:
+		return "mode";
 	default:
 		return NULL;
 	}
diff --git a/drivers/media/i2c/arducam-pivariety.h b/drivers/media/i2c/arducam-pivariety.h
index b2b8cbc47138..99d5ada309e8 100644
--- a/drivers/media/i2c/arducam-pivariety.h
+++ b/drivers/media/i2c/arducam-pivariety.h
@@ -52,6 +52,9 @@
 #define V4L2_CID_ARDUCAM_BASE		(V4L2_CID_USER_BASE + 0x1000)
 #define V4L2_CID_ARDUCAM_EXT_TRI	(V4L2_CID_ARDUCAM_BASE + 1)
 #define V4L2_CID_ARDUCAM_IRCUT		(V4L2_CID_ARDUCAM_BASE + 8)
+#define V4L2_CID_ARDUCAM_STROBE_SHIFT	(V4L2_CID_ARDUCAM_BASE + 14)
+#define V4L2_CID_ARDUCAM_STROBE_WIDTH	(V4L2_CID_ARDUCAM_BASE + 15)
+#define V4L2_CID_ARDUCAM_MODE		(V4L2_CID_ARDUCAM_BASE + 16)
 
 enum image_dt {
 	IMAGE_DT_YUV420_8 = 0x18,
-- 
2.44.0

